extras/gizmo/gizmo-transform.js

```diff
@@ -0,0 +1,364 @@
+            // point needs to be relative to gizmo for angle calculation
+            point.sub(gizmoPos);
+        }
+
+        if (isAllAxes) {
+            // calculate point distance from gizmo
+            tmpV1.copy(point).sub(gizmoPos).normalize();
+            tmpV2.copy(this.camera.up).add(this.camera.right).normalize();
+
+            const v = point.sub(gizmoPos).length() * tmpV1.dot(tmpV2);
+            point.set(v, v, v);
+        } else if (!isFacing) {
+            if (!isPlane && !isRotation) {
+                // reset normal based on axis and project position from plane onto normal
+                planeNormal.set(0, 0, 0);
+                planeNormal[axis] = 1;
+                tmpQ1.transformVector(planeNormal, planeNormal);
+                point.copy(planeNormal.scale(planeNormal.dot(point)));
+            }
+
+            // rotate point back to world coords
+            tmpQ1.invert().transformVector(point, point);
+
+            if (!isPlane && !isRotation) {
+                // set other axes to zero if not plane point
+                const v = point[axis];
+                point.set(0, 0, 0);
+                point[axis] = v;
+            }
+        }
+
+        let angle = 0;
+        if (isRotation) {
+            switch (axis) {
+                case 'x':
+                    angle = Math.atan2(point.z, point.y) * math.RAD_TO_DEG;
+                    break;
+                case 'y':
+                    angle = Math.atan2(point.x, point.z) * math.RAD_TO_DEG;
+                    break;
+                case 'z':
+                case 'face':
+                    angle = Math.atan2(point.y, point.x) * math.RAD_TO_DEG;
+                    break;
+            }
+        }
+        return { point, angle };
+    }
+
+    _drawGuideLine(pos, axis) {
+        tmpV1.set(0, 0, 0);
+        tmpV1[axis] = 1;
+        tmpV1.scale(GUIDELINE_SIZE);
+        tmpV2.copy(tmpV1).scale(-1);
+        tmpQ1.transformVector(tmpV1, tmpV1);
+        tmpQ1.transformVector(tmpV2, tmpV2);
+        this.app.drawLine(tmpV1.add(pos), tmpV2.add(pos), this._guideLineColor, true, this.layerGizmo);
+    }
+
+    _createMaterial(color) {
+        const material = new StandardMaterial();
+        material.emissive = color;
+        if (color.a !== 1) {
+            material.opacity = color.a;
+            material.blendType = BLEND_NORMAL;
+        }
+        return material;
+    }
+
+    // _createLight(angles) {
+    //     const light = new Entity('light');
+    //     light.addComponent('light', {
+    //         layers: [this.layerGizmo.id]
+    //     });
+    //     light.setEulerAngles(angles);
+    //     return light;
+    // }
+
+    _createTransform() {
+        // lighting
+        // const light = this._createLight(new Vec3(45, 0, -45));
+        // this.gizmo.addChild(light);
+
+        // center
+        this._center = new Entity('center');
+        this.gizmo.addChild(this._center);
+    }
+
+    set axisXColor(value) {
```

Comment:
This seems to take an array of numbers....but shouldn't it take a `Color`?

Reply:
The controls PCUI element for the color picker returns an array rather than `Color` hence why it is configured in this way.